plugins/transport-grpc/src/main/java/org/opensearch/transport/grpc/GrpcModulePlugin.java

```diff
@@ -0,0 +1,128 @@
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.env.Environment;
+import org.opensearch.env.NodeEnvironment;
+import org.opensearch.http.HttpServerTransport;
+import org.opensearch.plugins.NetworkPlugin;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.script.ScriptService;
+import org.opensearch.telemetry.tracing.Tracer;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.watcher.ResourceWatcherService;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static java.util.Collections.emptyList;
+import static org.opensearch.common.settings.Setting.intSetting;
+import static org.opensearch.common.settings.Setting.listSetting;
+
+/**
+ * Main class for the gRPC plugin
+ */
+public final class GrpcModulePlugin extends Plugin implements NetworkPlugin {
+    public static final Setting<Integer> SETTING_GRPC_PUBLISH_PORT = intSetting("grpc.publish_port", -1, -1, Setting.Property.NodeScope);
+
+    public static final Setting<List<String>> SETTING_GRPC_HOST = listSetting(
+        "grpc.host",
+        emptyList(),
+        Function.identity(),
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<List<String>> SETTING_GRPC_PUBLISH_HOST = listSetting(
+        "grpc.publish_host",
+        SETTING_GRPC_HOST,
+        Function.identity(),
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<List<String>> SETTING_GRPC_BIND_HOST = listSetting(
+        "grpc.bind_host",
+        SETTING_GRPC_HOST,
+        Function.identity(),
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> SETTING_GRPC_WORKER_COUNT = new Setting<>(
+        "grpc.netty.worker_count",
+        (s) -> Integer.toString(OpenSearchExecutors.allocatedProcessors(s)),
+        (s) -> Setting.parseInt(s, 1, "grpc.netty.worker_count"),
+        Setting.Property.NodeScope
+    );
+
+    private Netty4GrpcServerTransport transport;
+
+    @Override
+    public Collection<Object> createComponents(
+        Client client,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ResourceWatcherService resourceWatcherService,
+        ScriptService scriptService,
+        NamedXContentRegistry xContentRegistry,
+        Environment environment,
+        NodeEnvironment nodeEnvironment,
+        NamedWriteableRegistry namedWriteableRegistry,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        Supplier<RepositoriesService> repositoriesServiceSupplier
+    ) {
+        if (FeatureFlags.isEnabled(FeatureFlags.GRPC_ENABLE_SETTING) == false) {
+            throw new IllegalArgumentException("transport-grpc is experimental and feature flag must be enabled before use");
```

Comment:
This exception should only trigger when the `http.type` includes to `grcp`, installing the plugin without activation should cause no harm though

Reply:
gRPC feature flag removed.